File: Second Year (2013-2014)/OMGL7/TP3 - Website in JSP/src/db/AlbumDAO.java
package db;

import java.sql.*;
/**
* @author Andriamilanto Tompoariniaina - Canno Camille - Info2B
*/

/**
* Classe pour les objets DAO de Album
*/
public class AlbumDAO {
//Methodes
	/**
	*Constructeur
	*/
	public AlbumDAO() {}

	/**
	*Pour retrouver tous les tuples
	*/
	public ResultSet findAll() {
		Statement stat = null;
		String query = "";
		ResultSet ret = null;

		try {
			//Recuperation de la connexion
			Connection con = MySQLConnection.getInstance().getConnection();

			//Preparation de la requete en ligne
			stat = con.createStatement();

			//Preparation de la requete
			query = "SELECT * FROM Album;";

			//Le resultat a retourner
			ret = stat.executeQuery(query);
		}
		catch (SQLException e) {
			System.out.println("ERREUR: " + e.getMessage());
		}

		//Retourne l'execution de la requete sous la forme d'un objet ResultSet
		return ret;
	}

	/**
	*Permet de retrouver juste un tuple
	*@param id idAlbum de l'album a retrouver
	*/
	public Album find(int id) {
		Statement stat = null;
		String query = "";
		Album ret = new Album();

		try {
			//Recuperation de la connexion
			Connection con = MySQLConnection.getInstance().getConnection();

			//Preparation de la requete en ligne
			stat = con.createStatement();

			//Preparation de la requete
			query = "SELECT * FROM Album WHERE idAlbum = " + id + ";";

			//Retourne l'execution de la requete sous la forme d'un objet ResultSet
			ResultSet result = stat.executeQuery(query);

			//Si le resultat est bon, prends la premiere ligne
			if (result.first()) {
				ret.init(id, result.getString(2));
			}
		}
		catch(SQLException e) {
			System.out.println("ERREUR: " + e.getMessage());
		}

		return ret;
	}


	/**
    *Methode qui permet d'inserer un tuple
    *@param tuple Objet de type Album a inserer
    */
    public void insert(Album tuple) {
    	Statement stat = null;
        String query = "";

        //Recuperation des attributs de l'objet Album
        int idA = tuple.getIdAlbum();
        String lib = tuple.getLibelleAlbum();

        try {
	    	//Recuperation de la connexion
	        Connection con = MySQLConnection.getInstance().getConnection();

	        //Preparation de la requete
	        query = "INSERT INTO Album VALUES(" + idA + "," + lib + ");";

	        //Execute la requête
			stat.executeQuery(query);
		}
		catch(SQLException e) {
			System.out.println("ERREUR: " + e.getMessage());
		}
    }


    /**
    * Permet de faire la mise a jour d'un tuple
    *@param tuple Objet nouveau a mettre a la place de l'ancien
    */
    public void update(Album tuple) {
    	Statement stat = null;
        String query = "";

        //Recuperation des attributs de l'objet Album
        int idA = tuple.getIdAlbum();
        String lib = tuple.getLibelleAlbum();

        try {
	    	//Recuperation de la connexion
	        Connection con = MySQLConnection.getInstance().getConnection();

	        //Preparation de la requete
	        query = "UPDATE Album SET LibelleAlbum = " + lib + " WHERE IdAlbum = " + idA + ";";

	        //Execute la requête
			stat.executeQuery(query);
		}
		catch(SQLException e) {
			System.out.println("ERREUR: " + e.getMessage());
		}
    }


    /**
    * Permet de supprimer un tuple
    *@param idAlbum Id du tuple a supprimer
    */
    public void delete(int idAlbum) {
    	Statement stat = null;
        String query = "";

        try {
	    	//Recuperation de la connexion
	        Connection con = MySQLConnection.getInstance().getConnection();

	        //Preparation de la requete
	        query = "DELETE FROM Album WHERE IdAlbum = " + idAlbum + ";";

	        //Execute la requête
			stat.executeQuery(query);
		}
		catch(SQLException e) {
			System.out.println("ERREUR: " + e.getMessage());
		}
    }
}

File: Second Year (2013-2014)/OMGL7/TP2 - Website in PHP/musiconline/musiconline/db/AlbumDao.php
<?php
/* Authors: ANDRIAMILANTO Tompoariniaina && CANNO Camille
* Groupe: B1
* File: AlbumDao.php
*/

//Les importations nécessaires
import ('phpens.framework.Application');
import ('phpens.framework.db.DataObject');
import ('phpens.framework.db.DataAccessObject');
import ('musiconline.db.Album');


class AlbumDao implements DataAccessObject {
  //Attribut instance de DataAccessObject
  private static $dao;

  //Constructeur vide
  private function __construct() { }

  //Retourne l'instance d'AlbumDAO
  public final static function getInstance() {

    //Vérifie si l'instance est correcte ou non
    if(!isset(self::$dao)) {

      //S'il n'en est pas, il en créer un autre sans paramètres
      self::$dao= new AlbumDao();
    }

    //S'il en est un, il le retourne, sinon il retourne donc une instance vide
    return self::$dao;
  }


  //Retourne toute la table
  public final function findAll(){

     //Récupère la méthode de connection à la Base de donnée
     $dbc = Application::getInstance()->getDBConnection();

     //Variable où la requête est enregistrée
     $query = "select * from Album";

     //Requête en dur, prends la bdd et y exécute la requête
     $stmt = $dbc->query($query);

     /*Récupère tout les tuples de Album, et les transpose dans des instances de classe Album
     Et fini par enregistrer le résultat dans $results, FALSE est retourné s'ya erreur */
     $results = $stmt->fetchALL(PDO::FETCH_CLASS, 'Album');

     //Retourne le résultat de $results
     return $results;
   }


  //Permet d'ajouter un tuple
  public final function insert(DataObject $do){

    //Vérifie si l'objet entré en paramètre est instance de Album
    if($do instanceof Album){

      //Récupère la méthode de connection à la BDD
      $dbc = Application::getInstance()->getDBConnection();

      //Prépare la requête SQL
      $query = "insert into Album(IdAlbum,LibelleAlbum) values (:IdAlbum,:LibelleAlbum)";

      //Prépare la requête en dur avec la BDD et la requête correspondante
      $stmt = $dbc->prepare($query);

      //Remplace les paramètres ":IdAlbum" et "LibelleAlbum" par les valeurs de l'objet
      $stmt->bindValue(':IdAlbum',$do->getIDAlbum(),PDO::PARAM_STR);
      $stmt->bindValue(':LibelleAlbum',$do->getLIbelleAlbum(),PDO::PARAM_STR);

      //Exécute finallement la requête
      $stmt->execute($data);
    }
  }


  //Permet de mettre à jour un tuple
  public function update(DataObject $obj){

    //Vérifie si l'objet entré en paramètre est instance de Album
    if($obj instanceof Album){

      //Récupère la méthode de connection à la BDD
      $dbc = Application::getInstance()->getDBConnection();

      //Prépare la requête SQL
      $query = "UPDATE Album SET LibelleAlbum = :LibelleAlbum WHERE IdAlbum = :IdAlbum";

      //Prépare la requête en dur avec la BDD et la requête correspondante
      $stmt = $dbc->prepare($query);

      //Remplace les paramètres ":IdAlbum" et "LibelleAlbum" par les valeurs de l'objet
      $stmt->bindValue(':IdAlbum',$obj->getIDAlbum(),PDO::PARAM_STR);
      $stmt->bindValue(':LibelleAlbum',$obj->getLIbelleAlbum(),PDO::PARAM_STR);

      //Exécute finallement la requête
      $stmt->execute($data);
   }
  }


  //Permet de supprimer un tuple
  public function delete(DataObject $obj){

    //Vérifie si l'objet entré en paramètre est instance de Album
    if($obj instanceof Album){

      //Récupère la méthode de connection à la BDD
      $dbc = Application::getInstance()->getDBConnection();

      //Prépare la requête SQL
      $query = "DELETE FROM Album WHERE IdAlbum = :IdAlbum";

      //Prépare la requête en dur avec la BDD et la requête correspondante
      $stmt = $dbc->prepare($query);

      //Remplace les paramètres ":IdAlbum" par la valeur de l'objet
      $stmt->bindValue(':IdAlbum',$obj->getIDAlbum(),PDO::PARAM_STR);

      //Exécute finallement la requête
      $stmt->execute($data);
  }
}


//Creation d'une instance d'AlbumDAO
	$testAlbum = new AlbumDAO();

	//Test du findAll
	$findAll = $testAlbum->findAll();

	//Affichage du resultat
	echo $findAll;
?>
